backend/de.metas.business/src/main/java/de/metas/project/workorder/undertest/WorkOrderProjectObjectUnderTestRepository.java

```diff
@@ -139,12 +162,36 @@ public List<WOProjectObjectUnderTest> getByQuery(@NonNull final WOObjectUnderTes
 				.collect(ImmutableList.toImmutableList());
 	}
 
+	@Nullable
+	public WOProjectObjectUnderTest getByOrderLineId(@NonNull final OrderLineId orderLineId)
+	{
+		return Optional.ofNullable(queryBL.createQueryBuilder(I_C_Project_WO_ObjectUnderTest.class)
+										   .addEqualsFilter(I_C_Project_WO_ObjectUnderTest.COLUMNNAME_C_OrderLine_Provision_ID, orderLineId.getRepoId())
+										   .create()
+										   .firstOptional(I_C_Project_WO_ObjectUnderTest.class))
+				.filter(Optional::isPresent)
+				.map(Optional::get)
+				.map(WorkOrderProjectObjectUnderTestRepository::ofRecord)
+				.orElse(null);
```

Comment:
there is no unique index on `COLUMNNAME_C_OrderLine_Provision_ID` so no reason to only get the first match